### ISSUE TITLE
Could not find or load main class

### ISSUE THREAD

**Issue by kparsh19** [MasterDevX/Termux-Java#21]
Hi there. I had installed the package using your script. But when i issue the java command to run my program it throws the error "Could not find or load main class". I don't know how to solve this issue in termux and also -jshell command is not working. Can you help?

**Comment by GuzioMG** [comment]
I have no clue, what's causing the lack of `-jshell`, but "Could not find or load main class" is a problem with your application, not Java*. It means, that the file you're trying to run does not have a `main()` method. The only solution I can see here is:
- Java does not have access to your file (for some reason)
- Application is broken (probably)
- You're running a `.jar` file without `-jar` parameter (make sure it's the last one)

> *However, if you're still not sure, you can always try running a different app, that you're sure, it works. Minecraft server is running fine (although it melts your phone). I checked.

**Comment by kparsh19** [comment]
Well, I will give answer to your points in the same way
1) I think this is the main reason behind this. I have seen some videos about this same issue coming in windows, they have solved it by making a new environment variable
2) It is impossible (I hope) because it's termux
3) There is nothing related to .jar/-jar because it's one of the simplest program I have written using an editor with .java extension and I just want to compile/run it on termux
May be it is due to absence of -jshell or may be there's something wrong in this package. While compiling the program using javac (.java) it compiles fine and also I had seen that .class file is also generated but this error occurs when I try to run it using java command. I have checked the path.

**Comment by GuzioMG** [comment]
Well... It's probably exactly what you just did and I'm asking this question for no reason, but: Are you sure that (since you're running a `.class` file) you're using `java Program` not `java Program.class`? I just checked and `java Program.class` returns `Error: Could not find or load main class Program.class`.

**Comment by kparsh19** [comment]
Yeah, I have used java_program and not java_program.class with java command because I know that latter will create this error. But the unfortunate is the same error with java java_program command

**Comment by GuzioMG** [comment]
:-/ Well, I have no clue then.

**Comment by johnmalib** [comment]
According to the error message ("Could not find or load main class"), there are two categories of problems:

- The Main class could not be found
- The Main class could not be loaded

The Main class could not be found when there is a typo or wrong syntax in the fully qualified class name or it does not exist in the provided classpath.

The [Main](http://net-informations.com/java/basics/main.htm) class could not be loaded when the class cannot be initiated. Typically the main class extends another class and that class does not exist in the provided classpath.

For example:

`public class YourMain extends org.apache.camel.spring.Main`

If camel-spring is not included, this error will be reported.

Other reasons to occur this error is :

- File Extension
- Wrong package
- Invalid Classpath
- Wrong Class Name
